lib/assessment.js

```diff
@@ -99,25 +99,30 @@ module.exports = {
           // to make sure we correctly call endTransaction even
           // in the presence of errors.
           (callback) => {
-            var params = [
-              assessment_id,
-              user_id,
-              group_work,
-              authn_user_id,
-              mode,
-              time_limit_min,
-              date,
-            ];
-            sqldb.callWithClientOneRow(
-              client,
-              'assessment_instances_insert',
-              params,
-              (err, result) => {
-                if (ERR(err, callback)) return;
-                assessment_instance_id = result.rows[0].assessment_instance_id;
-                callback(null);
-              }
-            );
+            // if workspace, get workspace_id
+            if (!assessment_instance_id) {
```

Comment:
This condition will always be true since this variable isn't initialized to anything after it's declared a few lines up. What was this change supposed to handle, and what does the comment on the preceding line mean?

Reply:
This was part of a race condition fix from PR #2914 (when 2 students want to create a new instance simultaneously, if I understood correctly). This pairs with sprocs/assessment_instances_insert.sql in commit [24b1094](https://github.com/PrairieLearn/PrairieLearn/pull/5290/commits/24b1094a71ea807c295dc69c053c33eb0b010a7b) (I think I forgot to add it back in when I had to undo some commits because I managed to commit as Dev User from the PrairieLearn shell, oops) I copied over the comment from the original PR, but thinking it over, maybe should have skipped that.

Reply:
I understand why the variable is declared on line 94, my question is why there's the conditional on line 103. There is nothing between like 94 and line 103 that could set `assessment_instance_id` to anything, so AFAICT this conditional should be removed.

Reply:
Ah yes, I see. I believe this sorted itself out anyway with the prior merge conflict with the main branch I took care of in [21fc3a0](https://github.com/PrairieLearn/PrairieLearn/pull/5290/commits/21fc3a0ce87d5357c5f00c30bc415c1d96c06f3d)

Reply:
Ah, nice, looks great now!